program/src/state/config.rs

```diff
@@ -0,0 +1,52 @@
+use bytemuck::{Pod, Zeroable};
+use shank::ShankAccount;
+use solana_program::pubkey::Pubkey;
+use spl_discriminator::SplDiscriminate;
+use spl_pod::optional_keys::OptionalNonZeroPubkey;
+
+/// Configuration for a staking system.
+#[repr(C)]
+#[derive(Clone, Copy, Pod, ShankAccount, SplDiscriminate, Zeroable)]
+#[discriminator_hash_input("config::account")]
+pub struct Config {
+    /// Account disciminator.
+    ///
+    /// The discriminator is equal to `ArrayDiscriminator:: UNINITIALIZED` when
+    /// the account is empty, and equal to `Config::DISCRIMINATOR` when the account
+    /// is initialized.
+    pub discriminator: [u8; 8],
```

Comment:
This can maybe also be private, but up to you. I can't see a reason why someone would need to access the discriminator after the account has already been deserialized/unpacked.